index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
Yes, but sometimes in unexpected ways:

```js
'1-alpha' > '1' // true
'1.10' > '1.2' // false
```

Unless we want to go all-out with semver, I think we should limit the version to a single integer. I also think a single major version is enough to fulfill the requirements, as it's only used to signify breaking changes.